writer_objects.go

```diff
@@ -95,7 +98,18 @@ func (f *rsfWriter) writeStruct(v reflect.Value, tParent *tag, buf *bytes.Buffer
 	var totalSz int
 	for i := 0; i < v.NumField(); i++ {
 		t := &tag{}
-		skip, err := getTagInfo(v.Type(), i, t, tParent, v.Field(i).String())
+
+		// `fieldVal` is used for indexing arrays. We currently only support
+		// fixed strings and integers.
+		var fieldVal any
+		switch v.Field(i).Type().Kind() {
+		case reflect.String:
+			fieldVal = v.Field(i).String()
+		case reflect.Int, reflect.Int64, reflect.Int32, reflect.Int16, reflect.Int8:
+			fieldVal = v.Field(i).Int()
+		}
```

Comment:
Why not do this within `getTagInfo()`? Does it make it harder to test?

Reply:
Oh, good point. Let me look into that.

Reply:
I tried this, but it got me into a lot of refactoring to use `reflect.Value` instead of `reflect.Type`, which isn't compatible with all the scenarios. I think it's probably best to leave it as-is for now.